### PR TITLE
Adding basic version navigation

### PR DESCRIPTION
**Pull request by robscott**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
With v1alpha2 coming quickly, we need some kind of solution here. This is far from perfect, but it appears to work locally for me. I'm still very interested in a better long term solution for this. 

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

Note: This is closely related to @hbagdi's earlier PR: https://github.com/kubernetes-sigs/gateway-api/pull/757.

### CODE REVIEW

File: site-src/v1alpha2/index.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 ## What is the Gateway API?
 
+!!! danger "This version of the API has not been released yet."
```
Comment (robscott) [inline site-src/v1alpha2/index.md:3]: Had to duplicate the main page to distinguish between v1alpha1 and v1alpha2. At this point the only difference is the warning on the v1alpha2 page.

**Comment by robscott** [comment]
Deploy preview appears to work reasonably well: https://deploy-preview-778--kubernetes-sigs-gateway-api.netlify.app/

**Review by youngnick — COMMENTED** [review]
I'm working on something similar, but this will get us covered for now, this LGTM.

**Comment by robscott** [comment]
The main point of this PR was just to have a workaround in place so API reviewers could navigate v1alpha2 docs. I think @youngnick's upcoming updates are going to be much better, so I'm going to put a hold on this one and just use the deploy preview as a reference for API review.

/hold

**Comment by shaneutt** [comment]
created https://github.com/kubernetes-sigs/gateway-api/issues/783 as a follow-up to this to check all the pages that were noted as needing `v1alpha2` updates still :+1:

**Comment by robscott** [comment]
#782 was a much better solution and has merged in. We no longer need this as an interim solution.

/close

**Pull request closed without merging**